src/test/unit/assets/js/session/session-timeout.test.ts

```diff
@@ -0,0 +1,305 @@
+const expect = chai.expect;
+chai.use(require('sinon-chai'));
+
+describe('Session Timeout', () => {
+  let sandbox: sinon.SinonSandbox;
+  let sessionTimeout: SessionTimeout;
+  let extendSessionStub: sinon.SinonStub;
+  let addListenersStub: sinon.SinonStub;
+  let removeListenersStub: sinon.SinonStub;
+  let stopCountersStub: sinon.SinonStub;
+  let startCounterStub: sinon.SinonStub;
+  let resetModalMessageStub: sinon.SinonStub;
+  let signOutStub: sinon.SinonStub;
+  let openModalStub: sinon.SinonStub;
+  let startModalCountdownStub: sinon.SinonStub;
+  let restartCountersStub: sinon.SinonStub;
+  let clock: sinon.SinonFakeTimers;
+  let modalElement: HTMLElement;
+  let modalOverlayElement: HTMLElement;
+  let modalCountdownElement: HTMLElement;
+  let focusableElements: NodeListOf<Element>;
+  let firstFocusableElement: HTMLElement;
+  let lastFocusableElement: HTMLElement;
+  let bodyElement: HTMLElement;
+
+  beforeAll(() => {
+    document.body.innerHTML = `<div class="timeout-modal" id="timeout-modal">
+      <p id="dialog-description">the countdown</p>
+      <button id="extend-session">Extend</button>
+    </div>
+    <div class="modal-overlay" id="modal-overlay" tabindex="-1" aria-hidden="true"></div>`;
+    modalElement = document.querySelector('#timeout-modal');
+    modalOverlayElement = document.querySelector('#modal-overlay');
+    modalCountdownElement = document.querySelector('#dialog-description');
+    focusableElements = modalElement.querySelectorAll(
+      'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
+    );
+    firstFocusableElement = focusableElements[0] as HTMLElement;
+    lastFocusableElement = focusableElements[focusableElements.length - 1] as HTMLElement;
+    bodyElement = document.querySelector('body');
+    sessionTimeout = new SessionTimeout();
+  });
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+    clock = sandbox.useFakeTimers();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('init', () => {
+    it('should call addListeners and extend session', () => {
+      addListenersStub = sandbox.stub(sessionTimeout, 'addListeners');
+      extendSessionStub = sandbox.stub(sessionTimeout, 'extendSession');
+      sessionTimeout.init();
+      expect(addListenersStub.calledOnce);
+      expect(extendSessionStub.calledOnce);
+    });
+  });
+
+  describe('addListeners', () => {
+    it('should add click listener and call extendSession when click on button', () => {
+      extendSessionStub = sandbox.stub(sessionTimeout, 'extendSession');
+      const extendButton = document.querySelector('#extend-session');
+      sessionTimeout.addListeners();
+      const evt = new MouseEvent('click', {
+        view: window,
+        bubbles: true,
+        cancelable: true,
+        clientX: 20,
+      });
+      extendButton.dispatchEvent(evt);
+      expect(extendSessionStub.calledOnce);
```

Comment:
These expect checks does not assert anything gives code smell problem in sonar cloud